compiler/luci/import/src/Importer.cpp

```diff
@@ -126,6 +127,204 @@ void convert_graph(const luci::GraphBuilderSource &source, luci::CircleReader &r
+  auto circle_metadata = std::make_unique<luci::CircleImportMetadata>(reader);
+
+  // build a cache to identify if a tensor is output of an operator
+  // if this is set, we should not create a CircleConst for this tensor
+  for (uint32_t i = 0; i < operators.size(); ++i)
+  {
+    const auto op = operators[i];
+    assert(op != nullptr);
+    const auto outputs = luci::wrap(op->outputs());
+
+    for (uint32_t j = 0; j < outputs.size(); ++j)
+    {
+      auto tidx = outputs[j];
+      tensoroutputs->enroll(tidx);
+    }
+  }
+
+  // graph inputs; there are no input nodes in TFlite but just Tensors
+  // creating virtual input nodes will make possible to connect nodes that uses them
+  // all attributes of tensor should be copied to CircleInput node
+  for (const auto input : reader.inputs())
+  {
+    auto input_node = graph->nodes()->create<luci::CircleInput>();
+    assert(input_node != nullptr);
+    const auto tensor = tensors[input];
+    assert(tensor != nullptr);
+
+    luci::copy_tensor_attributes(tensor, input_node);
+    if (tensor->shape() == nullptr)
+      input_node->shape_status(luci::ShapeStatus::NOSHAPE);
+    else
+      input_node->shape_status(luci::ShapeStatus::VALID);
+
+    INFO(l) << "[luci] NodeFinder INPUT(" << input << ") = " << input_node << std::endl;
+    nodefinder->enroll(input, input_node);
+
+    // input_node is also an output to a tensor
+    tensoroutputs->enroll(input);
+
+    // Name
+    auto graph_input = graph->inputs()->create();
+    graph_input->name(input_node->name());
+
+    // Set GraphInputOutputIndex for graph
+    input_node->index(graph_input->index());
+
+    // Data type
+    graph_input->dtype(input_node->dtype());
+
+    const auto tensor_shape_signature = luci::wrap(tensor->shape_signature());
+    const auto tensor_shape = luci::wrap(tensor->shape());
+    assert(tensor_shape_signature.size() == 0 ||
+           tensor_shape_signature.size() == tensor_shape.size());
+
+    // Shape of GraphInput
+    auto input_shape = std::make_unique<loco::TensorShape>();
+    const auto &input_dims = tensor_shape; // in NHWC
+    input_shape->rank(input_dims.size());
+    for (uint32_t r = 0; r < input_dims.size(); ++r)
+    {
+      if (tensor_shape_signature.size() > 0 && tensor_shape_signature.at(r) == -1)
+        input_shape->dim(r).unset();
+      else
+        input_shape->dim(r).set(input_dims[r]);
+    }
+    graph_input->shape(std::move(input_shape));
+  }
+
+  // Create CircleConst nodes for constant tensors.
+  // NOTE Origin is intentionally not provided for constants.
+  for (uint32_t i = 0; i < tensors.size(); ++i)
+  {
+    luci::CircleNode *const_node = nullptr;
+
+    // create copy of const data if data is 'String' or copying requested by caller
+    if (copy_consts || luci::luci_datatype(tensors[i]->type()) == loco::DataType::STRING)
+      const_node = luci::create_circleconst(&gb_context, i);
+    else
+      const_node = luci::create_circle_const_ref(&gb_context, i);
```

Comment:
withour `ref` was `create_circleconst` and this is `create_circle_const` + `_ref` which, I hope you can follow as `create_circleconst_ref` for the name.